infra/bridge/src/controllers/hooks/index.js

```diff
@@ -0,0 +1,184 @@
+    const accessToken = await getTwitterOAuthAccessToken(
+      session.oAuthToken,
+      session.oAuthTokenSecret,
+      req.query.oauth_verifier
+    )
+    oAuthAccessToken = accessToken.oAuthAccessToken
+    oAuthAccessTokenSecret = accessToken.oAuthAccessTokenSecret
+  } catch (error) {
+    if (error.statusCode == 401) {
+      return res.status(401).send({
+        errors: ['The oauth-verifier provided is invalid.']
+      })
+    }
+    logger.error(error)
+    return res.status(500).send({
+      errors: ['Could not get a Twitter access token.']
+    })
+  }
+
+  let userProfileData
+  try {
+    userProfileData = await verifyTwitterCredentials(
+      oAuthAccessToken,
+      oAuthAccessTokenSecret
+    )
+  } catch (error) {
+    logger.error(error)
+    return res.status(500).send({
+      errors: ['Could not verify Twitter credentials.']
+    })
+  }
+
+  if (
+    userProfileData.username.toLowerCase() !==
+    process.env.TWITTER_ORIGINPROTOCOL_USERNAME.toLocaleLowerCase()
+  ) {
+    return res.status(400).send({
+      errors: ['Invalid account']
+    })
+  }
+
+  try {
+    await subscribeToHooks(oAuthAccessToken, oAuthAccessTokenSecret)
+  } catch (err) {
+    logger.error(err)
+    return res.status(400).send({
+      success: false,
+      errors: [`Failed to subscribe: ${err.message}`]
+    })
+  }
+
+  return res.status(200).send({
+    success: true
+  })
+})
+
+/**
+ * Webhook Authorization
+ */
+router.get('/twitter', (req, res) => {
+  const hmac = crypto
+    .createHmac('sha256', process.env.TWITTER_ORIGIN_CONSUMER_SECRET)
+    .update(req.query.crc_token)
+    .digest('base64')
+
+  res.status(200).send({
+    response_token: `sha256=${hmac}`
+  })
+})
+
+/**
+ * Twitter posts events to this endpoint
+ * Should always return 200 with no response
+ */
+router.post('/twitter', (req, res) => {
+  if (req.body.follow_events) {
+    // Follow event(s)
+    const events = req.body.follow_events
+    events.forEach(event => {
+      if (
+        event.target.screen_name ===
+        process.env.TWITTER_ORIGINPROTOCOL_USERNAME.toLocaleLowerCase()
+      ) {
+        // Store the follower in redis for 60 minutes
+        redisClient.set(
+          `twitter/follow/${event.source.id}`,
+          JSON.stringify(event),
+          'EX',
+          60 * 60 * 30
+        )
+      }
+    })
+  }
+
+  if (req.body.tweet_create_events) {
+    const events = req.body.tweet_create_events
+    events
+      .filter(event => {
+        // Ignore own tweets, retweets and favorites
+        return (
+          !event.retweeted &&
+          !event.favorited &&
+          event.user.screen_name !== process.env.TWITTER_ORIGINPROTOCOL_USERNAME
+        )
+      })
+      .forEach(event => {
+        // Note: Only the latest tweet will be in redis for 60 minutes
+        redisClient.set(
+          `twitter/share/${event.user.id}`,
+          JSON.stringify(event),
+          'EX',
+          60 * 60 * 30
+        )
+      })
+  }
+
```

Comment:
Might be useful to add a logger.info statement here with number of shares/follows that we added to redis. Always useful to have some logs to know what the process is doing.